pkg/controller/pod/pod_controller.go

```diff
@@ -240,6 +255,19 @@ func (r *ReconcilePod) checkIfK8sResourceExists(
 }
 
 func (r *ReconcilePod) createK8sObject(obj *unstructured.Unstructured) error {
```

Comment:
this is same as in line 284. we should try to reuse this function there to avoid duplication. same for other instances where we use *OC and *K8s objects. maybe embed *K8s into *OC struct to allow reuse (same in other controllers)? or maybe changing to interface can solve it?

Reply:
Removed the duplication, thanks.